internal/provider/waypoint/resource_waypoint_action.go

```diff
@@ -0,0 +1,607 @@
+	default:
+		return methodString, fmt.Errorf("unknown method")
+	}
+	return methodString, nil
+}
+
+func convertMethodToEnumType(method string) (waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethod, error) {
+	var methodEnum waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethod
+	switch method {
+	case "GET":
+		methodEnum = waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethodGET
+	case "POST":
+		methodEnum = waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethodPOST
+	case "PUT":
+		methodEnum = waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethodPUT
+	case "PATCH":
+		methodEnum = waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethodPATCH
+	case "DELETE":
+		methodEnum = waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethodDELETE
+	default:
+		return methodEnum, fmt.Errorf("unknown method")
+	}
+	return methodEnum, nil
+}
+
+func (r *ActionResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_waypoint_action"
+}
+
+func (r *ActionResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "The Waypoint Action resource managed the lifecycle of an Action.",
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The ID of the Action.",
+				Computed:    true,
+			},
+			"name": schema.StringAttribute{
+				Description: "The name of the Action.",
+				Required:    true,
+			},
+			"project_id": schema.StringAttribute{
+				Description: "The ID of the HCP project where the Action is located.",
+				Computed:    true,
+				Optional:    true,
+			},
+			"organization_id": schema.StringAttribute{
+				Description: "The ID of the HCP organization where the Action is located.",
+				Computed:    true,
+			},
+			// An Action description must be fewer than 125 characters if set.
+			"description": schema.StringAttribute{
+				Description: "A description of the Action.",
+				Optional:    true,
+			},
+			"request": schema.SingleNestedAttribute{
+				Description: "The kind of HTTP request this should trigger.",
+				Required:    true,
+				Attributes: map[string]schema.Attribute{
+					"custom": schema.SingleNestedAttribute{
+						Description: "Custom mode allows users to define the HTTP method, the request body, etc.",
+						Optional:    true,
+						Attributes: map[string]schema.Attribute{
+							"method": schema.StringAttribute{
+								Description: "The HTTP method to use for the request. Must be one of: 'GET', 'POST', 'PUT', 'DELETE', or 'PATCH'.",
+								Required:    true,
+								Validators: []validator.String{
+									stringvalidator.OneOf("GET", "POST", "PUT", "DELETE", "PATCH"),
```

Comment:
nit: If we're requiring/validating here, then the `convertMethodToStringType` and `convertMethodToEnumType` methods seem superfluous to me when we can convert them like so:
```golang
method := waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethod(plan.Request.Custom.Method.ValueString())
modelBody.ActionConfig.Request.Custom.Method = &method
```

later in the read we can just type cast I think

```golang
data.Request.Custom.Method = types.StringValue(string(*actionCfg.Request.Custom.Method))
```